analytics.go

```diff
@@ -453,12 +454,35 @@ func (c *client) send(msgs []message, retryAttempt int) {
 // Upload serialized batch message.
 func (c *client) upload(b []byte, targetNode string) error {
 	url := c.Endpoint + "/v1/batch"
-	req, err := http.NewRequest("POST", url, bytes.NewReader(b))
+	gzipPayload := func(data []byte) (io.Reader, error) {
+		var b bytes.Buffer
+		gz := gzip.NewWriter(&b)
+		_, err := gz.Write(data)
+		if err != nil {
+			return nil, err
+		}
+		if err = gz.Flush(); err != nil {
+			return nil, err
+		}
```

Comment:
Flush shouldn't be needed since
> Close closes the Writer by flushing any unwritten data to the underlying io.Writer and writing the GZIP footer. It does not close the underlying io.Writer.


```suggestion
```